### PR TITLE
feat(dynamic-sampling): Add post_delete and post_save hook to invalidate project config [TET-623]

### PR DESCRIPTION
**Pull request by iambriccardo**
This PR aims at implementing the project config invalidation after a release is `saved` or `deleted`. This implementation has the goal of keeping up to date rules with the state of the system.

### CODE REVIEW

File: tests/sentry/models/test_release.py [hunk 1]
```diff
@@ -1327,3 +1332,60 @@ def test_simple(self):
...
+        with Feature(
+            {
+                "organizations:dynamic-sampling": True,
+            }
+        ):
+            project = self.create_project(name="foo")
+            release = Release.objects.create(organization_id=project.organization_id, version="42")
+
+            # We store a boosted release for this project.
+            self.redis_client.hset(
```
Comment (andriisoldatenko) [inline tests/sentry/models/test_release.py:1383]: why do we need this line here?
you never assert it?

File: src/sentry/models/release.py [hunk 2]
```diff
@@ -67,6 +68,31 @@ class ReleaseCommitError(Exception):
...
+        from sentry.dynamic_sampling.feature_multiplexer import DynamicSamplingFeatureMultiplexer
+        from sentry.dynamic_sampling.latest_release_booster import ProjectBoostedReleases
+
+        ds_feature_multiplexer = DynamicSamplingFeatureMultiplexer(project)
+        project_boosted_releases = ProjectBoostedReleases(project.id)
+        # We want to invalidate the project config only if dynamic sampling is enabled and there exists boosted releases
+        # in the project.
+        if (
+            ds_feature_multiplexer.is_on_dynamic_sampling
+            and project_boosted_releases.has_boosted_releases()
```
Comment (andriisoldatenko) [inline src/sentry/models/release.py:83]: very good candidate for property, but unrelated to this PR.

Comment (andriisoldatenko) [inline src/sentry/models/release.py:83]:
```suggestion
            and project_boosted_releases.has_boosted_releases
```

File: tests/sentry/models/test_release.py (hunk 1 above)
Comment (iambriccardo) [reply tests/sentry/models/test_release.py:1383]: You are right, I need the assertion. It is needed to add a boosted release, but I will do that with the other class.

File: src/sentry/models/release.py (hunk 2 above)
Comment (iambriccardo) [reply src/sentry/models/release.py:83]: Will fix it, thanks!

**Review by andriisoldatenko — APPROVED** [review]

**Pull request merged**